Packet++/src/SomeIpLayer.cpp

```diff
@@ -326,10 +293,90 @@ bool SomeIpLayer::isSomeIpSdPacket(uint16_t serviceId, uint16_t methodId)
 	return serviceId == 0xFFFF && methodId == 0x8100;
 }
 
-bool SomeIpLayer::isSomeIpTpPacket(uint8_t msgType)
+bool SomeIpLayer::isDataValidSd(uint8_t *data, uint32_t dataLen)
+{
+	someiphdr *hdr = (someiphdr *)data;
+	return SomeIpLayer::isSomeIpSdPacket(hdr->serviceID, hdr->methodID);
+}
+
+// SomeIpTpLayer
+
+SomeIpTpLayer::SomeIpTpLayer(uint16_t serviceID, uint16_t methodID, uint16_t clientID, uint16_t sessionID,
+							 uint8_t interfaceVersion, MsgType type, uint8_t returnCode, uint32_t offset,
+							 bool moreSegmentsFlag, const uint8_t *const data, size_t dataLen)
+{
+	const size_t headerLen = sizeof(someiptphdr);
+
+	m_DataLen = headerLen + dataLen;
+	m_Data = new uint8_t[m_DataLen];
+	m_Protocol = SomeIP;
+	memset(m_Data, 0, headerLen);
+	memcpy(m_Data + headerLen, data, dataLen);
+
+	setServiceID(serviceID);
+	setMethodID(methodID);
+	setPayloadLength(dataLen + sizeof(uint32_t));
+	setClientID(clientID);
+	setSessionID(sessionID);
+	setProtocolVersion(0x01);
+	setInterfaceVersion(interfaceVersion);
+	setMessageType(type);
+	setReturnCode(returnCode);
+	setOffset(offset);
+	setMoreSegmentsFlag(moreSegmentsFlag);
+}
+
+bool SomeIpTpLayer::isSomeIpTpPacket(uint8_t msgType)
 {
 	/* Check if TP flag is set in msgType */
 	return (msgType & (uint8_t)SomeIpLayer::MsgType::TP_REQUEST) != 0;
 }
 
+bool SomeIpTpLayer::isDataValid(uint8_t *data, uint32_t dataLen)
+{
+	someiphdr *hdr = (someiptphdr *)data;
+	return SomeIpTpLayer::isSomeIpTpPacket(hdr->msgType);
+}
+
+uint8_t SomeIpTpLayer::setTpFlag(uint8_t messageType)
+{
+	return messageType | (uint8_t)SomeIpLayer::MsgType::TP_REQUEST;
+}

```

Comment:
Should this method be public?
I guess we need to use it in 2 places:
- `SomeIpTpLayer::computeCalculateFields()`
- SomeIpTpLayer::SomeIpTpLayer(...)`

Maybe we can make it private?